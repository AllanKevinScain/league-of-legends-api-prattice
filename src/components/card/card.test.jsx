import { render, screen } from '@testing-library/react';
import { Card } from '.';
const props = {
  championName: 'Aatrox',
  nickname: 'O a√ßougueiro de quinta',
};

describe('<Card />', () => {
  it('should render the card', () => {
    render(<Card {...props} />);

    const img = expect(screen.getByRole('img'));
    img.toBeInTheDocument();
    img.toHaveAttribute('alt', 'thumbnail');
    img.toHaveAttribute(
      'src',
      `https://ddragon.leagueoflegends.com/cdn/img/champion/splash/${props.championName}_0.jpg`,
    );

    let title = expect(screen.getByRole('heading', { name: `${props.championName} - ${props.nickname}` }));
    title.toBeInTheDocument();

    title = expect(screen.getByText(`${props.championName} - ${props.nickname}`));
    title.toBeInTheDocument();

    const link = expect(screen.getByRole('link'));
    link.toBeInTheDocument();
    link.toHaveAttribute('href', `https://www.leagueoflegends.com/pt-br/champions/${props.championName.toLowerCase()}`);
  });

  it('should match snpashot', () => {
    const { container } = render(<Card {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
