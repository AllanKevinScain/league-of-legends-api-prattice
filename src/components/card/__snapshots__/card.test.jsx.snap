// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should match snpashot 1`] = `
<div
  class="container-card"
>
  <img
    alt="thumbnail"
    class="thumbnail"
    src="https://ddragon.leagueoflegends.com/cdn/img/champion/splash/Aatrox_0.jpg"
  />
  <h3
    class="title"
  >
    Aatrox
     - 
    O açougueiro de quinta
  </h3>
  <span
    class="meet-link"
  >
    Conheça mais
     
    <a
      href="https://www.leagueoflegends.com/pt-br/champions/aatrox"
      rel="noreferrer"
      target="_blank"
    >
      @aqui
    </a>
  </span>
</div>
`;
